test/acceptance/tests/connect/connect_inject_namespaces_test.go

```diff
@@ -178,3 +182,120 @@ func TestConnectInjectNamespaces(t *testing.T) {
 		})
 	}
 }
+
+// Test the cleanup controller that cleans up force-killed pods.
+// These tests currently only test non-secure and secure without auto-encrypt installations
+// because in the case of namespaces there isn't a significant distinction in code between auto-encrypt
+// and non-auto-encrypt secure installations, so testing just one is enough.
+func TestConnectInjectNamespaces_CleanupController(t *testing.T) {
+	cases := []struct {
+		name                 string
+		destinationNamespace string
+		mirrorK8S            bool
+		secure               bool
+	}{
+		{
+			"single destination namespace",
+			staticClientNamespace,
+			false,
+			false,
+		},
+		{
+			"single destination namespace; secure",
+			staticClientNamespace,
+			false,
+			true,
+		},
+		{
+			"mirror k8s namespaces",
+			staticClientNamespace,
+			true,
+			false,
+		},
+		{
+			"mirror k8s namespaces; secure",
+			staticClientNamespace,
+			true,
+			true,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			cfg := suite.Config()
+			ctx := suite.Environment().DefaultContext(t)
+
+			helmValues := map[string]string{
+				"global.enableConsulNamespaces": "true",
+				"connectInject.enabled":         "true",
+				// When mirroringK8S is set, this setting is ignored.
+				"connectInject.consulNamespaces.consulDestinationNamespace": c.destinationNamespace,
+				"connectInject.consulNamespaces.mirroringK8S":               strconv.FormatBool(c.mirrorK8S),
+
+				"global.acls.manageSystemACLs": strconv.FormatBool(c.secure),
+				"global.tls.enabled":           strconv.FormatBool(c.secure),
+			}
+
+			releaseName := helpers.RandomName()
+			consulCluster := consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+
+			consulCluster.Create(t)
+
+			logger.Logf(t, "creating namespace %s", staticClientNamespace)
+			k8s.RunKubectl(t, ctx.KubectlOptions(t), "create", "ns", staticClientNamespace)
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+				k8s.RunKubectl(t, ctx.KubectlOptions(t), "delete", "ns", staticClientNamespace)
+			})
+
+			logger.Log(t, "creating static-client deployment")
+			staticClientOpts := &terratestk8s.KubectlOptions{
+				ContextName: ctx.KubectlOptions(t).ContextName,
+				ConfigPath:  ctx.KubectlOptions(t).ConfigPath,
+				Namespace:   staticClientNamespace,
+			}
+			k8s.DeployKustomize(t, staticClientOpts, cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-client-namespaces")
+
+			logger.Log(t, "waiting for static-client to be registered with Consul")
+			consulClient := consulCluster.SetupConsulClient(t, c.secure)
+			expectedConsulNS := staticClientNamespace
+			if !c.mirrorK8S {
+				expectedConsulNS = c.destinationNamespace
+			}
```

Comment:
Isn't `c.destinationNamespace` always equal to `staticClientNamespace`? Wouldn't the `expectedConsulNS` always be `staticClientNamespace`?

Reply:
Oops, yeah! Fixing...